community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexMap.java

```diff
@@ -95,32 +130,204 @@ public void foreachIndexProxy( BiConsumer<Long, IndexProxy> consumer )
         return indexesById.values();
     }
 
+    /**
+     * Get all indexes that would be affected by changes in the input labels and/or properties
+     * @param changedLabels set of labels that have changed
+     * @param unchangedLabels set of labels that are unchanged
+     * @param properties set of properties
+     * @return set of LabelSchemaDescriptors describing the potentially affected indexes
+     */
+    public Set<LabelSchemaDescriptor> getRelatedIndexes(
```

Comment:
I improved the javadoc to explain the guarantees on the method, but since the implementation is quite short the details can be left to the code.